python/ray/ml/tests/test_xgboost_predictor.py

```diff
@@ -0,0 +1,88 @@
+import os
+from ray.ml.predictors.integrations.xgboost import XGBoostPredictor
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.constants import MODEL_KEY, PREPROCESSOR_KEY
+import json
+
+import numpy as np
+import xgboost as xgb
+import tempfile
+
+
+class DummyPreprocessor(Preprocessor):
+    def transform_batch(self, df):
+        self._batch_transformed = True
+        return df * 2
+
+
+dummy_data = np.array([[1, 2], [3, 4], [5, 6]])
+dummy_target = np.array([0, 1, 0])
+model = xgb.XGBClassifier(n_estimators=10).fit(dummy_data, dummy_target).get_booster()
+
+
+def get_num_trees(booster: xgb.Booster) -> int:
+    data = [json.loads(d) for d in booster.get_dump(dump_format="json")]
+    return len(data) // 4
```

Comment:
Why divide by 4?

Reply:
this is due to how xgboost saves the model in json format